mautic/mautic
# #14119 Missing feature "Anonymize contact" / feature to add new event responsible to anonymize user data

- tests
- add field list type
- update lead events
- add helper anonymize
- change domain test.



| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴🟢
| Deprecations?                          | 🔴🟢
| BC breaks? (use the c.x branch)        | 🔴🟢
| Automated tests included?              | 🔴🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

Feature responsible to create new event action that can anonymize data. 
It's based on issue #14119 







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add new Contact with
    - Email
    - First Name
    - Last Name
    - Position
    - add tag 'Anonymize"
 
![mautic-issue-new-action-contact-form-1](https://github.com/user-attachments/assets/910b6f91-f530-4550-b6f8-f462ec628788)

3. Create a Segment name "_SegToAnonimyze_" with filter to add contacts with **Anonymize** tag
4. Run command `mautic:segments:update` to add contacts in segments
5. Create a Campaign based with segment SegToAnonimyze
6. Add new action with name "Anonymize User Data"
![Screenshot from 2024-10-07 14-08-16](https://github.com/user-attachments/assets/69e9fd33-6126-431d-8e6c-608a3b58e92e)

8. Fill form like:
    - Name : test
    - Execute this event : Immedialty
    - Pseudonymize data? Yes
    - Anonymize User Data from fields : Last Name
    - Delete User Data from fields : Position
    
![mautic-issue-new-action-form-edited](https://github.com/user-attachments/assets/30683fbc-14e1-4846-9b4f-199e1b49f59d)

8 - Save Campaign 
9 - Run command "mautic:campaigns:rebuild" to build campaign
10 - Run command  mautic:campaigns:trigger
11 - Come back to contact created and check:
    - Last name was change by hash
    - Position is empty

![mautic-issue-new-action-contact-form-edited](https://github.com/user-attachments/assets/f07e6f0f-3ade-498a-b808-05b20ddf3f03)
